### PR TITLE
Testing, Kotlin SDK

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
### Jira

- finally got JVM tests working properly
- added CI
- added page for opening a synced realm
- minor fixes related to kotlin/java sdk renames

### Staged Changes (Requires MongoDB Corp SSO)

- [Sync Data](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/kotlinCRUD/sdk/kotlin/app-services/sync-data/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/kotlin/shared/build.gradle.kts [hunk 1]
```diff
@@ -53,6 +55,13 @@ kotlin {
             iosX64Test.dependsOn(this)
             iosArm64Test.dependsOn(this)
         }
+        jvm().compilations["main"].defaultSourceSet {
+            dependencies {
+                implementation(kotlin("stdlib-jdk8"))
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-android:$coroutinesVersion")
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-test:$coroutinesVersion")
+            }
+        }
```
Comment (nathan-contino-mongo) [inline examples/kotlin/shared/build.gradle.kts:64]: this is the magic sauce that makes it all work. Even though KMM's common module effectively runs on the JVM... it defaults to running on android/ios instead. This little hack lets us run common tests right on the JVM, ensuring a nice speed boost and none of the tricky operating conditions of mobile emulators.

File: source/sdk/kotlin/app-services/sync-data.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
+.. _kotlin-sync-changes-between-devices:
+
+=========================================
```
Comment (cbush) [inline source/sdk/kotlin/app-services/sync-data.txt:3]: Why this change?

**Review by cbush — APPROVED** [review]

File: source/sdk/kotlin/app-services/sync-data.txt (hunk 2 above)
Comment (nathan-contino-mongo) [reply source/sdk/kotlin/app-services/sync-data.txt:3]: This is a page that I just... somehow forgot to include in the original PR. Part of some git cherry picking snafus that I went through when trying to productionize the PR. Very much a mistake on my part.
